qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -196,30 +218,34 @@ def write(self, cmd: str) -> None:
         """
         Select correct trace before querying
         """
-        self.root_instrument.active_trace(self.trace)
+        self.root_instrument.active_trace(self.trace_num)
         super().write(cmd)
 
     def ask(self, cmd: str) -> str:
         """
         Select correct trace before querying
         """
-        self.root_instrument.active_trace(self.trace)
+        self.root_instrument.active_trace(self.trace_num)
         return super().ask(cmd)
 
-    @staticmethod
-    def parse_paramstring(paramspec: str) -> Tuple[str, str, str]:
+    def parse_paramstring(self) -> Tuple[str, str]:
```

Comment:
this does not seem to be `parse_...` anymore - it now also gets the `paramspec`. Could you keep the separation for clarity?

Reply:
Agreed, I've removed this function and folded the functionality into the `_Sparam` function.